### ISSUE TITLE
optional compresion streaming 

### ISSUE THREAD

**Issue by MugurelGiurcanu** [XamlAnimatedGif/XamlAnimatedGif#26]
In case that an control contains large number of big animated gifs ( let's say more then 200, like in chat apps happens frequently ), and there is no much processor available, the animation stops. I assume that is caused by using compression streaming. ( because in WpfAnimatedGif, the applications works perfectly until it cracks ). So it will be an big plus if the XamlAnimatedGif let this option (compressed stream or not ) to the developer. Also if the control could rise an event when the controller is out of memory (to remove some gifs), would be more then enough.

**Comment by thomaslevesque** [comment]
I see the problem, but it would be very difficult to make it optional. It would require rewriting large parts of the library. Also, one of the main reason why I created XamlAnimatedGif was to solve the memory issues of WpfAnimatedGif; making this change would reintroduce the memory issues.

There are other improvements that could be done to solve your problem, though:
- reuse the same `Animator` when the same GIF is displayed multiple times (see #25)
- stop animating the image if it's not visible (not sure how hard that would be; it's probably not trivial to detect whether or not the image is actually visible)

**Comment by MugurelGiurcanu** [comment]
you have here a project with XamlAnimatedGif and XamlAnimatedGif with 2 kind of icons. One large, which baisically in my application will not be autoanimated, and small gif which will have autoplay flagg setted to True. 
https://drive.google.com/file/d/0B90UNI19qmavblltMFE3ZFh1X1k/view?pli=1

**Comment by thomaslevesque** [comment]
> With XamlAnimatedGif the UI will start freeze if i'm using more the 160 small images.

Each image is rendered separately, so the CPU has too much work to do. In my view, displaying so many animated GIFs at once has never been a realistic scenario, so I'm surprised you were able to put so many before it started freezing.

As I said before, #25 should improve this scenario when displaying the same GIF many times, since the rendering will only be done once.

**Comment by MugurelGiurcanu** [comment]
I can use more then 200 large animated gif in winforms without any problem.also in html I can do that. U're telling me in wpf, which should be much better then winforms, that's not possible? As u can see in my exambple after 5 large images the performance became doubtful. I can't continue any more using this library. sorry, man I can't risk my job for that.

**Comment by thomaslevesque** [comment]
> I can use more then 200 large animated gif in winforms without any problem.also in html I can do that. U're telling me in wpf, which should be much better then winforms, that's not possible?

Did I say that? I don't think so. Natively, WPF doesn't support animated GIFs, this is why this library exists. It's far from perfect, but it's better than nothing.

> As u can see in my exambple after 5 large images the performance became doubtful. I can't continue any more using this library. sorry, man I can't risk my job for that.

Are you trying to convince me of something? I'm doing my best to make this library as good as possible; I'm *not* trying to make everyone use it. If you don't like it, then don't use it; no need to apologize...

(as a side note, I really wouldn't want to use an app that shows 200 animated GIFs at once... but I guess that's a different issue)

**Comment by MugurelGiurcanu** [comment]
You should consider that feedback as an fowarded one. I have an
"competitor"(collegue) who say that WPF it's a wrong choise for writing a
chat. Skype is made in wpf, so it's posible. But the big issue was my
experience in wpf, which was zero. practicallt that's my first project in
WPF. I cope with lot of chanllenges, exept that one Animated Gif. The
actual chat on my company is in html, that's why I have so much presure on
animated gif, tiff, png, etc. Please don't take this personal, but the
situation is triky for me. If i don't came with an quick solution I'll be
in unpleasunt position. The first version was good until the number of
images became too big. The actual library have some problems after 5 gifs.
My app UI it's very loaded (and should be like that), Personally I don't
think that's mission imposible.

Sorry if my comment was too tough. I apologize for that .
Thank you for your efort of keep-ing me with hope for more then 6 months.

**Comment by thomaslevesque** [comment]
> You should consider that feedback as an fowarded one.

I do appreciate the feedback. As I said in my first reply, I have some ideas to improve performance. However, without native support for animated GIFs in WPF itself, it will be difficult to achieve the same performance as in Windows Forms or in a web browser. Maybe the LZW decoding could be made faster, and I'm sure there are more efficient ways to render the frames (perhaps using DirectX?), but I'm not an expert in this area... If you know a better way, I'd love to hear about it.

>  Sorry if my comment was too tough. I apologize for that .

Apology accepted; I may have overreacted myself, sorry about that.

**Comment by MugurelGiurcanu** [comment]
I do have some ideas how to improve that and I'll try to implement them. I could contribute to that project only if you'll accept my implementations. The main ideea is to keep property dependencies for Image control as it now. 

Thanks. 
​

**Comment by thomaslevesque** [comment]
> The main ideea is to keep property dependencies for Image control as it now.

Well, that's not something I intend to change anyway, since it would be a breaking change...

**Comment by yunushkin88** [comment]
Hello thomaslevesque!!! I loved Your library. But I don't know very well C#. Before that I wrote in C under microcontrollers. But I always try to write on C#. Maybe my question will seem trivial. Is it possible to change using your library Gif parametry of the picture. We have opened a gif and changed the frame rate in the gif or the background and closed the gif. The description says that it is possible or there is a view that only inside the program without saving in the settings file?
